actor/src/main/scala-3/org/apache/pekko/util/ByteString.scala

```diff
@@ -823,7 +874,33 @@ sealed abstract class ByteString
   override def indexWhere(p: Byte => Boolean, from: Int): Int = iterator.indexWhere(p, from)
 
   // optimized in subclasses
-  override def indexOf[B >: Byte](elem: B, from: Int): Int = indexOf(elem, from)
+  override def indexOf[B >: Byte](elem: B, from: Int): Int = super.indexOf(elem, from)
```

Comment:
I am aware that this is override is a bit weird, but:
- I think the old code was an infinite loop
- Removing this override breaks the MiMa checks

This should never be called anyway, but I think having it like this is a bit safer.